build_tools/build/src/ci_gen/job.rs

```diff
@@ -221,30 +225,55 @@ impl JobArchetype for VerifyLicensePackages {
 
 #[derive(Clone, Copy, Debug)]
 pub struct JvmTests {
-    pub graal_edition: graalvm::Edition,
+    pub graal_edition:   graalvm::Edition,
+    pub engine_launcher: engine::EngineLauncher,
 }
 
 impl JobArchetype for JvmTests {
     fn job(&self, target: Target) -> Job {
         let graal_edition = self.graal_edition;
+        let engine_launcher = self.engine_launcher;
         let job_name = format!("JVM Tests ({graal_edition})");
-        let test_results_dir = paths::ENSO_TEST_JUNIT_DIR
-            .get()
-            .ok()
-            .and_then(|buf| buf.to_str().map(|s| s.to_owned()))
-            .unwrap_or_else(|| "target/test-results/".to_owned());
-        let _upload_artifact_job = step::upload_artifact("Upload test results")
-            .with_custom_argument("name", format!("Test_Results_{}", target.0))
-            .with_custom_argument("path", test_results_dir);
         let mut job = RunStepsBuilder::new("backend test jvm")
-            .customize(move |step| vec![step, step::engine_test_reporter(target, graal_edition)])
+            .customize(move |step| {
+                let cleanup_engine_distribution = Step {
+                    run: Some(format!(
+                        "rm -rf {}",
+                        built_distribution_directories(engine_launcher)
+                    )),
+                    shell: Some(Shell::Bash),
+                    ..Default::default()
+                };
+
+                let download_engine_distribution =
+                    step::download_engine_distribution(target, engine_launcher, graal_edition);
+
+                let unpack_engine_distribution = Step {
+                    name: Some("Unpack Engine Distribution".into()),
+                    run: Some(
+                        "tar -xvf built-distribution.tar
+rm built-distribution.tar
+"
```

Comment:
Can we do something about this indentation? Not a big deal, but perhaps we can just use `\n` instead of multiline string?